### ISSUE TITLE
stuff that would be cool to have but I am not going to work on right now

### ISSUE THREAD

**Issue by patriciali** [patriciali/wingman#11]
1. view next scramble: After choosing a round and competitor, show judges the next scramble for the competitor. This would allow judges to verify the cube is scrambled correctly.

2. analytics: It would be cool to keep data such as when the timers are being used, and people's inspection times

3. how to distribute app: we want to be able to push out OTA updates.

**Comment by jbcm627** [comment]
4. Alarm feature.  We can call up competitors this way.  And make sure... staff... show up on time.

5. Chat integration, https://github.com/cubing/ccm/issues/188

**Comment by patriciali** [comment]
I think these are good ideas, but I don't know if wingman is the best place to put them.

first, the current vision is for it to be something only to be run at judging stations, and not for all competitors (of course they can install it if they want, but it's not going to be very useful for them). there's definitely a good argument for an app for competitors to check live results, be called up, etc., but that should probably be a separate app from wingman.

and second, I could be misguided on this, but I think at least in the short-term, wingman will be something that will run on dedicated phones owned by the organizer. the reason for this is because only a few types of phones will recognize frequencies outside of voice range, which we need to read stackmats (see http://www.jflei.com/2014/08/21/dialup-stackmat/). in order for judges to use their personal phones, we'd need a bunch of those modem devices. in the long-term I guess ideally we would have judges bring their own devices, because it's not practical for organizers in every area of the country to buy up 10 Galaxy Nexi, so maybe it's better to start on this sooner... I don't know. maybe @jfly has more to say.

what do you think?

**Comment by jbcm627** [comment]
> I think these are good ideas, but I don't know if wingman is the best place to put them.
Indeed, a separate app may be a better idea.

> only a few types of phones will recognize frequencies outside of voice range, which we need to read stackmats
I'm not sure what hardware has been looked at, but out of curiosity have (esp. dirt cheap) tablets been looked at, and/or sending a signal through another port?

**Comment by patriciali** [comment]
@jfly knows much more than I do

**Comment by jfly** [comment]
I agree with @patriciali here. In my mind's eye, wingman will only run on phones that organizers own. I do see the appeal of a dedicated "ccm app" that competitors would want to use. I don't have a strong opinion about where that should go, however. I would love to see wingman grow into such a thing (and only show its judging features if you're logged in with permission to judge the comp?), but it could also probably live pretty naturally as a separate app entirely.

A list of all devices I've tried out is available [here](http://www.jflei.com/stackmat-phones/). I would really love to run wingman on cheap tablets, so long as they're not so cheap as to be unusable. On thing I found when looking at tablets is that many tablets don't even *have* headset jacks, they only have headphone jacks. As for investigating another port, I focused investigation solely on the microphone jack, because that's the only thing that could possibly work across android, ios, and windows phones. Given that we're probably going to skip the modem entirely, building something android specific is totally okay. Daniel Hermansson actually was able to build a working stackmat to usb cable for android: https://www.speedsolving.com/forum/showthread.php?49357-Of-stackmats-and-smartphones&p=1016349&viewfull=1#post1016349 (getting this to go on iOS would probably be a nonstarter).

@patriciali, I promise that if we change the underlying hardware mechanism, [fskube](https://github.com/jfly/fskube) will be updated to speak it, which should keep wingman fairly agnostic to any of this.

**Comment by patriciali** [comment]
> I would love to see wingman grow into such a thing (and only show its judging features if you're logged in with permission to judge the comp?)

that's a good idea! because all the competitors will have CCM accounts anyway, for registering for comps, right?

**Comment by jfly** [comment]
Yep! We're not gonna allow people to register without an account.
